pkg/controller/subscription/subscription_controller.go

```diff
@@ -73,13 +83,74 @@ func (r *ReconcileSubscription) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
-	if instance.Spec.InstallPlanApproval != v1alpha1.ApprovalManual {
-		instance.Spec.InstallPlanApproval = v1alpha1.ApprovalManual
-		err = r.client.Update(context.TODO(), instance)
+	if subscription.Spec.InstallPlanApproval != operatorsv1alpha1.ApprovalManual {
+		subscription.Spec.InstallPlanApproval = operatorsv1alpha1.ApprovalManual
+		err = r.client.Update(context.TODO(), subscription)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
 	}
 
-	return reconcile.Result{}, nil
+	result, err := r.HandleUpgrades(context.TODO(), subscription)
+
+	return result, err
+}
+
+func (r *ReconcileSubscription) HandleUpgrades(ctx context.Context, rhmiSubscription *operatorsv1alpha1.Subscription) (reconcile.Result, error) {
+	if !rhmiConfigs.IsUpgradeAvailable(rhmiSubscription) {
+		logrus.Infof("no upgrade available")
+		return reconcile.Result{}, nil
+	}
+
+	logrus.Infof("RHMI upgrade available")
+
+	latestRHMIInstallPlan, err := rhmiConfigs.GetLatestInstallPlan(ctx, rhmiSubscription, r.client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	latestRHMICSV, err := rhmiConfigs.GetCSV(latestRHMIInstallPlan)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	config := &integreatlyv1alpha1.RHMIConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      installation.DefaultCloudResourceConfigName,
+			Namespace: r.operatorNamespace,
+		},
+	}
+	err = r.client.Get(ctx, k8sclient.ObjectKey{Name: config.Name, Namespace: config.Namespace}, config)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = rhmiConfigs.UpdateStatus(ctx, r.client, config, latestRHMIInstallPlan)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	canUpgradeNow, err := rhmiConfigs.CanUpgradeNow(config)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !rhmiConfigs.IsUpgradeServiceAffecting(latestRHMICSV) || canUpgradeNow {
+		eventRecorder := r.mgr.GetEventRecorderFor("RHMI Upgrade")
+		err = rhmiConfigs.ApproveUpgrade(ctx, r.client, latestRHMIInstallPlan, eventRecorder)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// Requeue the reconciler until the RHMI subscription upgrade is complete
+		return reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: 10 * time.Second,
+		}, nil
+	}
+	logrus.Infof("not automatically upgrading a Service Affecting Release")
+	return reconcile.Result{
+		Requeue:      true,
+		RequeueAfter: 10 * time.Second,
+	}, nil
```

Comment:
Should this requeue every 10 seconds when we don't want to upgrade yet? Would it be better to increase the time?

Reply:
good point, updated to once a minute